File: output/LLaMa/avatar/Java/Python/codeforces_242_B.py
import math

INF = (1e9 + 5)
def main():
    n = int(input())
    a = []
    b = []
    left = INF
    right = 0
    for _ in range(n):
        a.append(int(input()))
        b.append(int(input()))
        left = min(left, a[-1])
        right = max(right, b[-1])
    for i in range(n):
        if left == a[i] and right == b[i]:
            print(i + 1)
            return
    print(-1)

if __name__ == "__main__":
    main()

File: output/LLaMa/avatar/Python/Java/codeforces_242_B.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    static BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
    static StringTokenizer tokenizer = new StringTokenizer("");

    static int nextInt() throws IOException {
        while (!tokenizer.hasMoreTokens()) {
            tokenizer = new StringTokenizer(reader.readLine());
        }
        return Integer.parseInt(tokenizer.nextToken());
    }

    public static void main(String[] args) throws IOException {
        int n = nextInt();
        int id = 0;
        int l = nextInt();
        int r = nextInt();
        for (int i = 1; i < n; i++) {
            int li = nextInt();
            int ri = nextInt();
            if (li <= l && r <= ri) id = i;
            else if (li < l || r < ri) id = n;
            l = Math.min(l, li);
            r = Math.max(r, ri);
        }
        System.out.println(id == n ? -1 : (id + 1));
    }
}